.github/workflows/doc-tests.yaml

```diff
@@ -92,3 +94,24 @@ jobs:
 
       - name: Test the docs build
         run: cd $GITHUB_WORKSPACE/docs && yarn install && yarn build
+
+  stylecheck:
+    name: Lint docs prose style
```

Comment:
How long does this job take to run? We could decide not to run it on merge group since the prose check has to pass on the PR anyway.

Also, can we add an `if: ` condition here to skip it completely if `needs.changes.outputs.changed` is true?

Reply:
Vale should take a few seconds to 30s for a large docs set and comprehensive rule set (https://github.com/errata-ai/vale?tab=readme-ov-file#benchmarks). Based on a [test workflow run](https://github.com/gravitational/teleport/actions/runs/7226227065/job/19691339196) from a few weeks ago, setting up the job and checking out `gravitational/teleport` add around 10 seconds to that.

I'll add the `if` condition and look into not running it for merge groups!